### PR TITLE
lokalize: init at 22.12.3

### PR DESCRIPTION
**Pull request by niknetniko**
###### Description of changes

Adds Lokalize. Quoting the application website:

> Lokalize is the localization tool for KDE software and other free and open source software. It is also a general computer-aided translation system (CAT) with which you can translate OpenDocument files (*.odt). Translate-Toolkit is used internally to extract text for translation from .odt to .xliff files and to merge translation back into .odt file.

See https://apps.kde.org/en/lokalize/.

Previously attempted at #89743.
Also works towards #62168.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -9967,6 +9967,13 @@
       fingerprint = "E576 BFB2 CF6E B13D F571  33B9 E315 A758 4613 1564";
     }];
   };
+  niknetniko = {
```
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:9970]: `maintainer-list` should be in a dedicated commit (in this same pull request).

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by BenediktBroich** [comment]
Result of `nixpkgs-review pr 202740` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>libsForQt512.lokalize</li>
    <li>libsForQt514.lokalize</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>libsForQt5.lokalize (libsForQt515.lokalize ,plasma5Packages.lokalize)</li>
  </ul>
</details>

**Comment by BenediktBroich** [comment]
When clicking `OpenDocument translate` i do get the error `Install package "translate-toolkit" and try again`.

Some of the listed  [dependencies](https://invent.kde.org/sdk/lokalize/-/blob/master/PACKAGING) for this application are also missing. Does the Application need these dependencies?

**Comment by niknetniko** [comment]
> Some of the listed [dependencies](https://invent.kde.org/sdk/lokalize/-/blob/master/PACKAGING) for this application are also missing. Does the Application need these dependencies?

Right, I missed these since I only needed gettext to work.

It would be probably best to include all of these, but e.g. `translate-toolkit` doesn't seem to be packaged yet for NixOS, and I'm not familiar with those, so it will take a while to get these included.

File: pkgs/applications/kde/lokalize.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+, qtbase
+, kactivities
+, kross
+, qtscript
+, hunspell
+, gettext
+}:
+
+mkDerivation rec {
+  pname = "lokalize";
```
Comment (AndersonTorres) [inline pkgs/applications/kde/lokalize.nix:13]: Where is version, src etc?

**Comment by niknetniko** [comment]
The OpenDocument parts of Lokalize still use Python 2, so I am not inclined to add support for those.

**Review by AndersonTorres — APPROVED** [review]

**Comment by niknetniko** [comment]
This should be redone for Plasma 6.

**Pull request closed without merging**